### PR TITLE
Add migration tool for mm_forum

### PR DESCRIPTION
**Pull request by martin-helmich**

### CODE REVIEW

File: Classes/Service/InstallService.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+     */
+    protected $messageQueueByIdentifier = '';
+
+    /**
+     * Initializes the install service
+     */
+    public function __construct()
+    {
+
+        $this->messageQueueByIdentifier = 'extbase.flashmessages.tx_extensionmanager_tools_extensionmanagerextensionmanager';
```
Comment (martin-helmich) [inline Classes/Service/InstallService.php]: This is a constant value; why not just make this the default value (l. 48)?

File: Classes/Service/InstallService.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+     * Adds a Flash Message to the Flash Message Queue
+     *
+     * @param FlashMessage $flashMessage
+     * @return void
+     */
+    protected function addFlashMessage(FlashMessage $flashMessage)
+    {
+        if ($flashMessage) {
+            /** @var $flashMessageService \TYPO3\CMS\Core\Messaging\FlashMessageService */
+            $flashMessageService = GeneralUtility::makeInstance('TYPO3\\CMS\\Core\\Messaging\\FlashMessageService');
```
Comment (martin-helmich) [inline Classes/Service/InstallService.php]: Can we retrieve the FlashMessageService via DI?

File: Classes/Service/Migration/UserMigrationService.php [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        return $this->generateOutput();
+    }
+
+    /**
+     * @param $pid
+     * @return bool|\mysqli_result|object
+     */
+    protected function getUsers($pid)
+    {
+        return $this->databaseConnection->exec_SELECTquery('*', $this->getOldTableName(), 'pid = ' . $pid);
```
Comment (martin-helmich) [inline Classes/Service/Migration/UserMigrationService.php]: `$pid` comes directly from `$_POST["users"]`. This should be escaped, even if the eventual form is only accessible from the back-end.

File: class.ext_update.php [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        return $output;
+    }
+
+    /**
+     * @return string
+     */
+    protected function renderForm()
+    {
+        /* @var $partial \TYPO3\CMS\Fluid\View\StandaloneView */
+        $partial = $this->objectManager->get('TYPO3\CMS\Fluid\View\StandaloneView');
```
Comment (martin-helmich) [inline class.ext_update.php]: Use `::class` consistently?

File: Classes/Service/InstallService.php (hunk 2 above)
Comment (kpurrmann) [inline Classes/Service/InstallService.php]: I think there was a reason why I implemented it this way but I'm not sure. Think we should keep it like this, but use ::class constant.

**Pull request merged**